### PR TITLE
Notes

### PR DESCRIPTION
**Pull request by vijaytembugade**
Added Note sidebar
Added Editor
Added Nested Routes

### CODE REVIEW

File: src/Components/AllNotes/AllNotes.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          </div>
+        </div>
+      )}
+
+      <div dangerouslySetInnerHTML={{ __html: value }}></div>
+    </div>
+  );
+};
+
+export default AllNotes;
```
Comment (horsemaker) [inline src/Components/AllNotes/AllNotes.js:65]: You could separate out the components here.

File: src/Components/Header/Header.js [hunk 2]
```diff
@@ -1,47 +1,74 @@
...
 function Header() {
+  const {
+    authState: { isLoggedIn },
+  } = useAuth();
+
+  const navMenuForbiddenPaths = ["/", "/login", "/signup"];
+
+  const { pathname } = useLocation();
+
+  console.log(pathname);
```
Comment (horsemaker) [inline src/Components/Header/Header.js:17]: You could remove this `console.log`

File: src/Components/NoteEditor/NoteEditor.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const NoteEditor = () => {
+  const [value, setValue] = useState("");
+  return (
+    <>
+      <div className="note-editor">
+        <div className="title title-md secondary-text">Create New Note</div>
+        <input className="note-title" type="text" placeholder="Title of Note" />
+        <div className="new-note">
+          <Quill value={value} setValue={setValue} />
```
Comment (horsemaker) [inline src/Components/NoteEditor/NoteEditor.js:13]: I guess this should be `ReactQuill`

File: src/Components/Quill/Quill.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      ["clean"],
+    ],
+  };
+
+  return (
+    <ReactQuill
+      className="quill-editor"
+      theme="snow"
+      modules={modules}
+      // formats={formats}
```
Comment (horsemaker) [inline src/Components/Quill/Quill.js:23]: comment

**Review by horsemaker — APPROVED** [review]

File: src/Components/SignupForm/SignupForm.js [hunk 5]
```diff
@@ -1,27 +1,99 @@
...
+    confirmPassword: "",
+  });
+
+  const { email, password, confirmPassword } = user;
+
+  const handleSignUp = async (e) => {
+    e.preventDefault();
+    try {
+      if (
+        email === "" ||
```
Comment (harshkotkar5557) [inline src/Components/SignupForm/SignupForm.js:26]: you can use required attributes to validate input field should not be empyt

File: src/Contexts/authContext/auth-context.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <AuthContext.Provider value={{ authState, authDispatch }}>
+      {children}
+    </AuthContext.Provider>
+  );
+};
+
+const useAuth = () => {
+  const context = useContext(AuthContext);
+  if (context === undefined) {
```
Comment (harshkotkar5557) [inline src/Contexts/authContext/auth-context.js:26]: if(!context) will work same as if(context === undefined)

**Review by harshkotkar5557 — APPROVED** [review]

**Pull request merged**